### PR TITLE
Update clinical data status of public files during release build

### PR DESCRIPTION
**Pull request by joneubank**
It was found during release process that some files had out of date status for `has_clinical_data` property, so I have added a step to the release build process that will check in with the clinical service for every file that will be made public to ensure that their clinical data status is up to date.

This property is meant to be updated in the file manager DB in response to a kafka message that the donor was updated, however at the moment these clinical update messages are not being sent or not being received correctly in some environments. This change will ensure that during the (infrequent) data release process we can go through our public files and ensure they are listed correctly.

## Additional Updates
- Moved prettier config out of `package.json` and into a dedicated config file.
- Made the prettier config use tabs

### CODE REVIEW

File: src/services/fileManager.ts [hunk 1]
```diff
@@ -49,43 +49,49 @@ import Logger from '../logger';
...
-  }
-
-  return file;
+/**
+ * Update File record from RDPC data and commit those changes to the file repo.
+ * @param file
+ * @param rdpcFileDocument
+ * @returns
+ */
+export async function updateFileFromRdpcData(file: File, rdpcFileDocument: RdpcFileDocument): Promise<File> {
```
Comment (joneubank) [inline src/services/fileManager.ts:78]: Renamed this function to better reflect its purpose. The `fileManager.ts` file was re-formatted to the updated prettier rules as a result.

File: src/services/release.ts [hunk 2]
```diff
@@ -104,127 +104,137 @@ export async function calculateRelease(): Promise<void> {
...
+
+		// 3a. Update public files with latest clinical data status
+		const { results: updatedExpectedPublicFiles } = await PromisePool.withConcurrency(5)
+			.for(expectedPublicFiles)
+			.handleError((error, file) => {
+				logger.error(`Failed to update clinical data status in DB for ${file}: ${error}`);
+			})
+			.process(async file => {
+				return await fileManager.updateFileClinicalData(file);
+			});
```
Comment (joneubank) [inline src/services/release.ts:170]: This is the only code change made to `services/release.ts`. This is the work required to update the clinical data for every file that will be made public. It is being done with some minimal concurrency to try to speed up the process. All the actual work is performed by the existing function `fileManager.updateFileClinicalData`.

File: .prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+	"printWidth": 120,
+	"trailingComma": "all",
+	"singleQuote": true,
+	"useTabs": true
```
Comment (joneubank) [inline .prettierrc.json:5]: Tabs is the new standard. This code base will correct to tabs over time with this change.

File: package.json [hunk 4]
```diff
@@ -1,135 +1,130 @@
...
-    "winston": "^3.2.1",
-    "yamljs": "^0.3.0",
-    "zod": "^3.22.2"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "pretty-quick --staged"
-    }
-  },
-  "prettier": {
```
Comment (justincorrigible) [inline package.json:130]: Appreciate the separation of concerns 🙏

File: tslint.json [hunk 5]
```diff
@@ -1,43 +1,43 @@
...
-    "no-trailing-whitespace": true,
-    "no-null-keyword": true,
-    "prefer-const": true,
-    "jsdoc-format": true
-  }
+	"rules": {
+		"class-name": true,
+		"comment-format": [true, "check-space"],
+		"no-async-without-await": true,
+		"indent": [false, "tabs"],
```
Comment (justincorrigible) [inline tslint.json:6]: what does "false" do here?

Comment (joneubank) [reply tslint.json:6]: make it not do

Comment (joneubank) [reply tslint.json:6]: and when not do, linting passes

Comment (joneubank) [reply tslint.json:6]: I'll run a chore and update all files to re-enable linting with tabs as the default after this work is released.

**Review by justincorrigible — APPROVED** [review]
LGTM 👍

**Pull request merged**